titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/direct/DefaultTaskToPodConverter.java

```diff
@@ -114,7 +122,8 @@ public V1Pod apply(Job<?> job, Task task) {
                 .schedulerName("default-scheduler")
                 .containers(Collections.singletonList(container))
                 .terminationGracePeriodSeconds(POD_TERMINATION_GRACE_PERIOD_SECONDS)
-                .restartPolicy(NEVER_RESTART_POLICY);
+                .restartPolicy(NEVER_RESTART_POLICY)
+                .affinity(podAffinityFactory.buildV1Affinity(job, task));
```

Comment:
should we add this only if `isOwnedByKubeScheduler()`? Any harm in adding it when tasks are being scheduled by Fenzo? (i.e.: will it negatively impact other controllers?)

Reply:
This check is made earlier. This code is executed only with direct (non Fenzo) scheduling.